src/main/java/uk/gov/companieshouse/company/metrics/processor/CompanyMetricsProcessor.java

```diff
@@ -96,5 +97,26 @@ private String extractCompanyNumber(String resourceUri) {
         return null;
     }
 
+    private void handleResponse(
+            final HttpStatus httpStatus,
+            final String logContext,
+            final Map<String, Object> logMap)
+            throws NonRetryableErrorException, RetryableErrorException {
+        logMap.put("status", httpStatus.toString());
+        if (HttpStatus.BAD_REQUEST == httpStatus) {
+            // 400 BAD REQUEST status is not retry-able
+            String message = "400 BAD_REQUEST response received from company-metrics-api";
+            logger.errorContext(logContext, message, null, logMap);
+            throw new NonRetryableErrorException(message);
+        } else if (httpStatus.is4xxClientError() || httpStatus.is5xxServerError()) {
```

Comment:
Can you please check these if conditions in charges delta consumer

Reply:
in Charges delta consumer i could see - !httpStatus.is2xxSuccessful(). Do you want me to change it to be inline with charges consumer? I am thinking if any 3xx or 1xx status codes they all fall under the retry.

Reply:
As discussed updated the condition.